### PR TITLE
[Feature]support create table with mapType column

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>
up to now, we just create table with map/struct type in external table, but not in starrocks native table. 

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 24 (95.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|7|85.71%|[5452]|
|:large_blue_circle:|com/starrocks/analysis/ColumnDef.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/TypeDef.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/analysis/TypeDef.java [hunk 1]
```diff
@@ -92,6 +94,19 @@ public void analyze() throws AnalysisException {
...
 
+    public void analyze(boolean isOlap) throws AnalysisException {
+        if (isOlap && (!FeConstants.runningUnitTest)) {
+            // we haven't support create table with map or struct type column in native table
+            Type innerType = Type.getInnermostType(parsedType);
+            if (innerType.isMapType() || innerType.isStructType()) {
+                throw new AnalysisException("Unsupported data type: " + parsedType.toSql());
+            }
+            analyze();
+        } else {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/TypeDef.java:105]: put analyze() to outside, there is no need to write else branch.

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/StructTypeTest.java [hunk 2]
```diff
@@ -23,7 +23,9 @@
 import com.starrocks.catalog.StructField;
 import com.starrocks.catalog.StructType;
 import com.starrocks.catalog.Type;
+import com.starrocks.common.FeConstants;
 import com.starrocks.utframe.StarRocksAssert;
+// import org.junit.AfterClass;
```
Comment (Youngwb) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/StructTypeTest.java:28]: delete this

File: fe/fe-core/src/main/java/com/starrocks/analysis/TypeDef.java (hunk 1 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/analysis/TypeDef.java]: done

**Review by Smith-Cruise — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|7|85.71%|[5452]|
|:large_blue_circle:|com/starrocks/analysis/ColumnDef.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/TypeDef.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Pull request merged**